functions.php

```diff
@@ -56,6 +59,22 @@ function twentytwentytwo_styles() {
 
 endif;
 
+/**
+ * Create variables linked to global styles.
+ *
+ * @return string
+ */
+function twentytwentytwo_theme_variables() {
+	if ( ! function_exists( 'gutenberg_get_global_styles' ) ) {
+		return;
+	}
+
+	// Get the properties of the button that we want to use in other blocks like search and file.
+	$button_styles = gutenberg_get_global_styles( array(), 'core/button' );
+
+	// Create variables that point to the value of those customizable properties.
+	return ':root, body { --wp--theme--button--color--text: ' . $button_styles['color']['text'] . '; --wp--theme--button--color--background: ' . $button_styles['color']['background'] . '; --wp--theme--button--border--radius: ' . $button_styles['border']['radius'] . '; }';
```

Comment:
FWIW I'm hesitant to add a workaround like this to the theme. Open to other opinions and feedback.

Reply:
Yeah, this feels just a bit too complicated for my liking. Since this is the sort of thing that should most definitely be handled natively by Gutenberg at some point, I lean towards not patching it in the theme. 😕

I'm not too concerned about the file block, but the Search block seems like a higher priority since we're actually using that in a template.